File: 1.0/src/cz/mg/vulkan/PFNvkBindImageMemory2KHR.java
package cz.mg.vulkan;

public class PFNvkBindImageMemory2KHR extends VkFunctionPointer {
    public PFNvkBindImageMemory2KHR() {
    }

    protected PFNvkBindImageMemory2KHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkBindImageMemory2KHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkBindImageMemory2KHR(long value) {
        setValue(value);
    }

    public PFNvkBindImageMemory2KHR(VkInstance instance) {
        super(instance, new VkString("vkBindImageMemory2KHR"));
    }

    public void call(VkDevice device, VkUInt32 bindInfoCount, VkBindImageMemoryInfoKHR pBindInfos, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), bindInfoCount != null ? bindInfoCount.getVkAddress() : VkPointer.getNullAddressNative(), pBindInfos != null ? pBindInfos.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void call(long vkaddress, long device, long bindInfoCount, long pBindInfos, long rval);
}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkBindImageMemory2KHR.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkBindImageMemory2KHR_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong bindInfoCount, jlong pBindInfos, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkBindImageMemory2KHR f = (PFN_vkBindImageMemory2KHR)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        *((uint32_t*)jniLongToPointer(bindInfoCount)),
        ((VkBindImageMemoryInfo*)jniLongToPointer(pBindInfos))
    );
}


jint Java_cz_mg_vulkan_PFNvkBindImageMemory2KHR_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jint bindInfoCount, jlong pBindInfos){
    (void)env;
    (void)jc;
    PFN_vkBindImageMemory2KHR f = (PFN_vkBindImageMemory2KHR)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        (uint32_t)bindInfoCount,
        ((VkBindImageMemoryInfo*)jniLongToPointer(pBindInfos))
    );
}

